src/libraries/System.Numerics.Vectors/tests/GenericVectorTests.cs

```diff
@@ -846,6 +846,56 @@ public void VectorSingleEqualsNaNTest()
             var nan = new Vector<float>(float.NaN);
             Assert.True(nan.Equals(nan));
         }
+
+        [Fact]
+        public void VectorDoubleEqualsNonCanonicalNaNTest()
+        {
+            var maxSignificand = (1UL << 53) - 1;
+            // NaN with mantissa bits set
+            var snan = BitConverter.UInt64BitsToDouble(BitConverter.DoubleToUInt64Bits(double.NaN) | maxSignificand);
```

Comment:
I think this is actually wrong for `snan`

This will end up "or'ing" `0x1F_FFFF_FFFF_FFFF` which is 1 too many bits, but also has the MSB of the significand set, where-as we want it "clear" for `sNaN`.

Basically any value between `0x7FF8_0000_0000_0000` and `0x7FFF_FFFF_FFFF_FFFF` is `+qNaN`
Any value between `0x7FF0_0000_0000_0001` and `0x7FF7_FFFF_FFFF_FFFF` is `+sNaN`

Reply:
oops. good catch. fixed.